### PR TITLE
Scraping logic included in the new refactored Twitter API code

### PR DESCRIPTION
**Pull request by Mira-Alf**
## Modifications:
* Changes copied from main branch latest to the new branch:
  -  cli.js: send to email option code was modified
  - send-to-kindle.js: colors were used in the successful email console log line
  - helpers.js: validateEmail logic was copied over
  - puppeteer index was copied over to the older location as well as the scraping folder inside the twitter api
* Changes related to the Twitter API for puppeteer
  - twindle-cli/index.js was modified to call the new puppeteer code
  - twitter\twitter.js: a new function was added called to process an array of tweet ids returned by scraping project
  - twitter/transformations/tweets-array-endpoint.js: processes the multiple ids api hit response

## Things to do
* If tweet older than 7 days hook up puppeteer scraping project
* Test all flags/options in CLI application  and remove the -p flag finally

## Attach Screenshot

> Note 2 code reviewer approval needed. Approach in twitter group & discord channel.

### CODE REVIEW

**Comment by Mira-Alf** [comment]
Big shoutout to @johnjacobkenny for his huge accomplishment in the twitter API refactoring - code definitely looks much better.

File: twindle-cli/twitter/scraping/index.js [hunk 1]
```diff
@@ -1 +1,90 @@
...
+    defaultViewport: {
+      width,
+      height,
+    },
+    args: [`--force-device-scale-factor=${factor}`, `--window-size=${width},${height}`],
+  });
+
+  const context = browser.defaultBrowserContext();
+  context.clearPermissionOverrides();
+  context.overridePermissions("https://twitter.com", ["clipboard-read", "clipboard-write"]);
```
Comment (proful) [inline twindle-cli/twitter/scraping/index.js:27]: @Mira-Alf please check the main branch again. The scrapping logic part improved by @PuruVJ 

We no need to use clipboard read write which is not very clean.

Comment (johnjacobkenny) [reply twindle-cli/twitter/scraping/index.js:27]: Yes @PuruVJ please merge in your changes to this branch.

Comment (PuruVJ) [reply twindle-cli/twitter/scraping/index.js:27]: Will do boss! 😁

Comment (PuruVJ) [reply twindle-cli/twitter/scraping/index.js:27]: Actually they're already merged! I forgot to remove permissions override. Doesn't matter much, but I'll remove it in my next commit.

Great work here @johnjacobkenny @Mira-Alf. We're on the right track

**Comment by johnjacobkenny** [comment]
@PuruVJ I think Mira actually has done most of the stuff. I think just the index from puppeteer in `twindle-cli/twitter/scraping` needs updating. But you just take a look to ensure its all there anyways.

**Comment by johnjacobkenny** [comment]
Once you're good to go, we can delete everything in main, and just merge in the new branch.

**Comment by PuruVJ** [comment]
@johnjacobkenny It's the latest code already. That part that Proful pointed out is left over artifact from the copy paste method.

**Pull request merged**